src/base/OpenMCProblemBase.C

```diff
@@ -355,7 +367,7 @@ OpenMCProblemBase::externalSolve()
     _console << " Skipping running OpenMC as the mesh has not changed!" << std::endl;
     return;
   }
-
+  openmc::settings::n_particles = nParticles();
```

Comment:
Do you need this line? The code already had support for changing the number of particles each batch (for Dufek-Gudowski relaxation). Please check if the infrastructure can be combined somehow